### ISSUE TITLE
TSDB status API served using remote write port

### ISSUE THREAD

**Issue by yeya24** [thanos-io/thanos#6239]
**What happened**:

Thanos Receiver serves TSDB status API `/api/v1/status/tsdb` using its remote write address, by default it is `remote-write.address`.

**What you expected to happen**:

I expect that HTTP API is served using Receiver's http address specified with `--http-address` flag, not the port for remote write.

Actually it is not very clear to me why we have a dedicated `remote-write.address` address for remote write only. Ideally we should consolidate it with the other http server.

**Comment by fpetkovski** [comment]
Good catch. I agree we should have only one http server and use the same port for everything.

**Comment by yeya24** [comment]
Thanks for the quick reply. This could be a breaking change and I am not sure what kind of approach we should take.
I guess we would still keep the remote write address, but we can start serving the TSDB status API in both addresses?

**Comment by fpetkovski** [comment]
Agree, this will be a pretty big change, so we need to support both servers/ports at the same time for at least a few releases. We can emit a deprecation warning in the logs in addition to adding a changelog entry.

**Comment by jnyi** [comment]
We found the top metrics cardinality is very useful info to track users' usage, we want a trending view of it instead of instant, so how about making this API (or create another API) which has prometheus metrics page format instead of json blob so we can scrape it?
